package/MDAnalysis/guesser/base.py

```diff
@@ -0,0 +1,205 @@
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""
+Base guesser classes --- :mod:`MDAnalysis.guesser.base`
+================================================================
+
+Derive topology reader classes from the base class in this module. All
+topology readers raise :exc:`IOError` upon failing to read a topology
+file and :exc:`ValueError` upon failing to make sense of the read data.
+
+Classes
+-------
+
+.. autoclass:: GuesserBase
+   :members:
+   :inherited-members:
+
+"""
+from .. import _GUESSERS
+import numpy as np
+from ..  import _TOPOLOGY_ATTRS
+import logging
+from typing import Dict
+
+logger = logging.getLogger("MDAnalysis.guesser.base")
+
+
+class _GuesserMeta(type):
+    """Internal: guesser classes registration
+
+    When classes which inherit from GuesserBase are *defined*
+    this metaclass makes it known to MDAnalysis.  'context'
+    attribute  are read:
+     - `context` defines the context of the guesser class for example: forcefield specific context
+     as MartiniGuesser and file specific context as PDBGuesser.
+
+    Eg::
+
+      class FooGuesser(GuesserBase):
+          format = 'foo'
+
+  """
+    def __init__(cls, name, bases, classdict):
+        type.__init__(type, name, bases, classdict)
+       
+        _GUESSERS[classdict['context'].upper()] = cls
+     
+
+class GuesserBase(metaclass=_GuesserMeta):
+    """Base class for context-aware guessers
+
+    Parameters
+    ----------
+    universe : Universe, optional
+        Supply a Universe to the guesser.  This then become the source of atom attributes
+        to be used in guessing processes (more relevant with the universe guess_topologyAttributes API).
+    **kwargs: to pass additional data to the guesser that can be used with different methos. 
+    Eg van der Waals radii that is used with bond guessing methods
+ 
+   """
+    context = 'base'
+    _guesser_box : Dict = {}
+
+    def __init__(self, universe=None, **kwargs):
+        self._universe = universe
+        self._kwargs = kwargs
+
+    @property
+    def universe(self):
+        return self._universe
+    
+    @universe.setter
+    def set_universe(self, u):
+        self._universe = u
```

Comment:
I think there is a risk associated with allowing to change the universe associated with a universe. It makes it possible to have a mismatch between the guesser and the universe. I would remove the setter and make the attribute read-only.